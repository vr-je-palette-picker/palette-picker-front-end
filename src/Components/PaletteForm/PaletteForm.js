import React, { Component } from 'react';
import './PaletteForm.scss';

export class PaletteForm extends Component {
  constructor() {
    super();
    this.state = {
      paletteName: '',
      project: ''
    };
  }

  render() {
    return (
      <div className='PaletteForm'>
        <h2 className='PaletteForm__h2--title'>Save New Palette</h2>
        <label
          className='PaletteForm__label--palette-name'
          htmlFor='paletteName'
        >
          Palette Name:
        </label>
        <input
          className='PaletteForm__input--palette-name'
          type='text'
          placeholder='Palette Name'
          value={this.state.paletteName}
        ></input>
        <label className='PaletteForm__label--project' htmlFor='project'>
          Choose A Project:
        </label>
        <select className='PaletteForm__select--project'>
          {/* map through projects in App state */}
        </select>
      </div>
    );
  }
}
