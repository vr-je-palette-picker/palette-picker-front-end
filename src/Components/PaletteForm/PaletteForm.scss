.PaletteForm {
  align-items: center;
  background-color: white;
  display: flex;
  flex-direction: column;
  font-family: 'Dosis', sans-serif;
  height: 80%;
  justify-content: space-evenly;
}

.PaletteForm__h2--title,
.PaletteForm__label--palette-name,
.PaletteForm__input--palette-name,
.PaletteForm__label--project,
.PaletteForm__select--project,
.PaletteForm__button--cancel,
.PaletteForm__button--save {
  font-size: 30px;
  height: 51px;
  margin: 5px;
  width: 100%;
}

.PaletteForm__h2--title {
  font-size: 50px;
}

.PaletteForm__select--project,
.PaletteForm__button--cancel,
.PaletteForm__button--save {
  cursor:pointer;
}