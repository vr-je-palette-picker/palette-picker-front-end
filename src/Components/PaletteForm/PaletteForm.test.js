import React from 'react';
import { shallow } from 'enzyme';
import PaletteForm from './PaletteForm';

describe('PaletteForm', () => {
  let wrapper;
  let projects = [{}, {}, {}];
  let close = jest.fn();
  let newPalette = jest.fn();
  let findProjectByName = jest.fn();

  beforeEach(() => {
    wrapper = shallow(
      <PaletteForm
        projects={projects}
        close={close}
        newPalette={newPalette}
        findProjectByName={findProjectByName}
      />
    );
  });

  it('should match the snapshot', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
