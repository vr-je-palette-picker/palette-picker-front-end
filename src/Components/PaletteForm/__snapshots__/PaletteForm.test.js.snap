// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PaletteForm should match the snapshot 1`] = `
<div
  className="PaletteForm"
>
  <h2
    className="PaletteForm__h2--title"
  >
    Save New Palette
  </h2>
  <label
    className="PaletteForm__label--palette-name"
    htmlFor="paletteName"
  >
    Palette Name:
  </label>
  <input
    className="PaletteForm__input--palette-name"
    name="paletteName"
    onChange={[Function]}
    placeholder="Palette Name"
    type="text"
    value=""
  />
  <label
    className="PaletteForm__label--project"
    htmlFor="project"
  >
    Choose A Project:
  </label>
  <select
    className="PaletteForm__select--project"
    onChange={[Function]}
    value="Please select a project"
  >
    <option
      value=""
    >
      Please select
    </option>
    <option
      key="0"
    />
    <option
      key="1"
    />
    <option
      key="2"
    />
  </select>
  <button
    className="PaletteForm__button--cancel"
    onClick={[Function]}
    type="button"
  >
    Cancel
  </button>
  <button
    className="PaletteForm__button--save"
    onClick={[Function]}
    type="button"
  >
    Save
  </button>
</div>
`;
